File: front/app/types/physical/port_slot.ts
import * as v from 'valibot'

// PortSlot
export const portSlotSchema = v.object({
    id: v.string(),
    name: v.pipe(v.string(), v.nonEmpty()),
    description: v.nullish(v.string()),
    portTypeId: v.nullish(v.string()),
})

export const portSlotFormSchema = v.partial(portSlotSchema, ['id'])

// Export des types
export type PortSlotType = v.InferInput<typeof portSlotSchema>
export type PortSlotTypeForm = v.InferInput<typeof portSlotFormSchema>

File: api/app/graphql/app/physical/types/port_slot.rb
# frozen_string_literal: true

module App
  module Physical
    module Types
      # PortSlotType
      class PortSlot < ::Types::BaseObject
        graphql_name 'PortSlotType'

        field :id, ID, null: true
        field :name, String, null: false
        field :port_type_id, ID, null: true
        field :description, String, null: true
      end
    end
  end
end
